test/tabs/test-firefox-tabs.js

```diff
@@ -1191,10 +1194,63 @@ exports.testTabDestroy = function(assert, done) {
   })
 };
 
+// related to bug 942511
+// https://bugzilla.mozilla.org/show_bug.cgi?id=942511
+exports['test active tab properties defined on popup closed'] = function (assert, done) {
+    let defaultOpenWindowPref = getPref(OPEN_IN_NEW_WINDOW_PREF);
+    let defaultPopupPref = getPref(DISABLE_POPUP_PREF);
+
+    setPref(OPEN_IN_NEW_WINDOW_PREF, 2);
+    setPref(DISABLE_POPUP_PREF, false);
+
+    let tabID = "";
+    let popupClosed = false;
+
+    tabs.open({
+      url: 'about:blank',
+      onReady: function (tab) {
+        tabID = tab.id;
+        tab.attach({
+          contentScript: 'var popup = window.open("about:blank");' +
+                         'popup.close();'
+        });
+        
+        windowObserver.once('close', () => {
+          popupClosed = true;
+        });
+
+        windowObserver.on('activate', () => {
+          // Only when the 'activate' event is fired after the popup was closed.
+          if(popupClosed) {
+            popupClosed = false;
+            let activeTabID = tabs.activeTab.id;
+            if (activeTabID) {
+                assert.equal(tabID, activeTabID, 'ActiveTab properties are correct');
+            }
+            else {
+              assert.fail('ActiveTab properties undefined on popup closed');
+            }
+            setPref(OPEN_IN_NEW_WINDOW_PREF, defaultOpenWindowPref);
+            setPref(DISABLE_POPUP_PREF, defaultPopupPref);
+            tab.close(done);
+          }
+        });
+      }
+    });
+};
+
+before(exports, (name, assert) => resetPopupPrefs);
```

Comment:
Do we really need to do this before each test? You're already doing it after each test so it seems redundant.